utilities/network.py

```
@@ -745,28 +746,47 @@ def assert_ping_successful(
 
     Returns:
         str or None: First found valid IP version address, or None.
     """
     if not (vm or virt_handler_pod):
         raise ValueError("must send VM or virt-handler pod")
 
-    if vm:
-        addr_list = vm.vmi.interfaces[0]["ipAddresses"]
-    else:
-        addr_list = [ip_addr["ip"] for ip_addr in virt_handler_pod.instance.status.podIPs]
+    def _resolve_ip():
```

Ruff (0.15.7)

[warning] Missing return type annotation for private function `_resolve_ip`

(ANN202)